core/client/src/blockchain.rs

```diff
@@ -37,6 +50,13 @@ pub trait HeaderBackend<Block: BlockT>: Send + Sync {
 	/// Get block hash by number. Returns `None` if the header is not in the chain.
 	fn hash(&self, number: NumberFor<Block>) -> Result<Option<Block::Hash>>;
 
+	/// Get light header, usually from an in-memory cache, without hitting database.
+	/// Returns `None` if the header is not in the chain.
+	fn get_light_header(&self, id: BlockId<Block>) -> Result<Option<LightHeader<Block>>>;
```

Comment:
convention is not to call this `get`. also `light_header` is a confusing name

Reply:
Would you prefer  `cached_header` or `header_data`?

Reply:
`CachedHeaderMetadata` perhaps